diff --git a/arch/arm/plat-omap/mailbox.c b/arch/arm/plat-omap/mailbox.c
index 08a2df7..16030aa 100644
--- a/arch/arm/plat-omap/mailbox.c
+++ b/arch/arm/plat-omap/mailbox.c
@@ -298,6 +298,8 @@ static int omap_mbox_startup(struct omap_mbox *mbox)
 
 static void omap_mbox_fini(struct omap_mbox *mbox)
 {
+	tasklet_kill(&mbox->txq->tasklet);
+	flush_work(&mbox->rxq->work);	
 	mbox_queue_free(mbox->txq);
 	mbox_queue_free(mbox->rxq);
 
