diff --git a/drivers/dsp/bridge/core/tiomap3430.c b/drivers/dsp/bridge/core/tiomap3430.c
index c6c3e18..fdd5ef8 100644
--- a/drivers/dsp/bridge/core/tiomap3430.c
+++ b/drivers/dsp/bridge/core/tiomap3430.c
@@ -288,7 +288,7 @@ static dsp_status bridge_brd_monitor(struct bridge_dev_context *hDevContext)
 			PWRDM_POWER_ON, OMAP3430_IVA2_MOD, OMAP2_PM_PWSTCTRL);
 		/* Set the SW supervised state transition */
 		(*pdata->dsp_cm_write)(OMAP34XX_CLKSTCTRL_FORCE_WAKEUP,
-					OMAP3430_IVA2_MOD, CM_CLKSTCTRL);
+					OMAP3430_IVA2_MOD, OMAP2_CM_CLKSTCTRL);
 
 		/* Wait until the state has moved to ON */
 		while ((*pdata->dsp_prm_read)(OMAP3430_IVA2_MOD, OMAP2_PM_PWSTST) &
@@ -296,7 +296,7 @@ static dsp_status bridge_brd_monitor(struct bridge_dev_context *hDevContext)
 			;
 		/* Disable Automatic transition */
 		(*pdata->dsp_cm_write)(OMAP34XX_CLKSTCTRL_DISABLE_AUTO,
-					OMAP3430_IVA2_MOD, CM_CLKSTCTRL);
+					OMAP3430_IVA2_MOD, OMAP2_CM_CLKSTCTRL);
 	}
 	(*pdata->dsp_prm_rmw_bits)(OMAP3430_RST2_IVA2, 0,
 					OMAP3430_IVA2_MOD, RM_RSTCTRL);
@@ -596,7 +596,7 @@ static dsp_status bridge_brd_start(struct bridge_dev_context *hDevContext,
 
 /*CM_CLKSTCTRL_IVA2 = 0x00000003 -To Allow automatic transitions */
 		(*pdata->dsp_cm_write)(OMAP34XX_CLKSTCTRL_ENABLE_AUTO,
-					OMAP3430_IVA2_MOD, CM_CLKSTCTRL);
+					OMAP3430_IVA2_MOD, OMAP2_CM_CLKSTCTRL);
 
 		/* Let DSP go */
 		dev_dbg(bridge, "%s Unreset\n", __func__);
@@ -678,7 +678,7 @@ static dsp_status bridge_brd_stop(struct bridge_dev_context *hDevContext)
 			PWRDM_POWER_OFF, OMAP3430_IVA2_MOD, OMAP2_PM_PWSTCTRL);
 		/* Set the SW supervised state transition for Sleep */
 		(*pdata->dsp_cm_write)(OMAP34XX_CLKSTCTRL_FORCE_SLEEP,
-					OMAP3430_IVA2_MOD, CM_CLKSTCTRL);
+					OMAP3430_IVA2_MOD, OMAP2_CM_CLKSTCTRL);
 	} else {
 		clk_status = dsp_clk_disable(DSP_CLK_IVA2);
 	}
