diff --git a/drivers/dsp/bridge/core/tiomap3430.c b/drivers/dsp/bridge/core/tiomap3430.c
index 8129cba..c6c3e18 100644
--- a/drivers/dsp/bridge/core/tiomap3430.c
+++ b/drivers/dsp/bridge/core/tiomap3430.c
@@ -285,7 +285,7 @@ static dsp_status bridge_brd_monitor(struct bridge_dev_context *hDevContext)
 		/* IVA2 is not in ON state */
 		/* Read and set PM_PWSTCTRL_IVA2  to ON */
 		(*pdata->dsp_prm_rmw_bits)(OMAP_POWERSTATEST_MASK,
-			PWRDM_POWER_ON, OMAP3430_IVA2_MOD, PM_PWSTCTRL);
+			PWRDM_POWER_ON, OMAP3430_IVA2_MOD, OMAP2_PM_PWSTCTRL);
 		/* Set the SW supervised state transition */
 		(*pdata->dsp_cm_write)(OMAP34XX_CLKSTCTRL_FORCE_WAKEUP,
 					OMAP3430_IVA2_MOD, CM_CLKSTCTRL);
@@ -675,7 +675,7 @@ static dsp_status bridge_brd_stop(struct bridge_dev_context *hDevContext)
 		/* IVA2 is not in OFF state */
 		/* Set PM_PWSTCTRL_IVA2  to OFF */
 		(*pdata->dsp_prm_rmw_bits)(OMAP_POWERSTATEST_MASK,
-			PWRDM_POWER_OFF, OMAP3430_IVA2_MOD, PM_PWSTCTRL);
+			PWRDM_POWER_OFF, OMAP3430_IVA2_MOD, OMAP2_PM_PWSTCTRL);
 		/* Set the SW supervised state transition for Sleep */
 		(*pdata->dsp_cm_write)(OMAP34XX_CLKSTCTRL_FORCE_SLEEP,
 					OMAP3430_IVA2_MOD, CM_CLKSTCTRL);
