diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index d02af55..c539fb4 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -411,6 +411,7 @@ static int __init omap3_beagle_i2c_init(void)
 {
     omap_register_i2c_bus(1, 2600, beagle_i2c1_boardinfo,
             ARRAY_SIZE(beagle_i2c1_boardinfo));
+    omap_register_i2c_bus(2, 400, NULL, 0);
     /* Bus 3 is attached to the DVI port where devices like the pico DLP
      * projector don't work reliably with 400kHz */
     omap_register_i2c_bus(3, 100, NULL, 0);
